include/deal.II/base/function_lib.h

```diff
@@ -1254,6 +1254,81 @@ namespace Functions
      */
     const Table<dim,double>                     data_values;
   };
+
+
+  /**
+   * A class that represents a function object for a polynomial. A polynomial
+   * is composed by multiple monomials,i-e. in 1-d they have the form
+   * $x^\alpha + x^\beta + ... + x^\epsilon$, in 2-d the form
+   * $x_1^{\alpha_1}x_2^{\alpha_2} + x_1^{\beta_1}x_2^{\beta_2} + ... +
+   * x_1^{\epsilon_1}x_2^{\epsilon_2}$, and in 3-d
+   * $x_1^{\alpha_1}x_2^{\alpha_2}x_3^{\alpha_3} + x_1^{\beta_1}x_2^{\beta_2}x_3^{\beta_3}
+   * + ... + x_1^{\epsilon_1}x_2^{\epsilon_2}x_3^{\epsilon_3}$.
+   * The class's constructor takes a Table<2,double> to describe the set of
+   * exponents and a Vector<double> to describe the set of coefficients.
+   *
+   *  @author Ángel Rodríguez, 2015
+   */
+  template <int dim>
+  class Polynomial : public Function<dim>
+  {
+  public:
+    /**
+     * Constructor. The coefficients and the exponents of the polynomial are
+     * passed as arguments. The Table<2, double> exponents has a number of rows
+     * equal to the number of monomials of the polynomial and a number of columns
+     * equal to dim. The i-th row of the table exponents contains the
+     * $\alpha_1...\alpha_dim$ exponents of the i-th monomial
+     * $x_1^{\alpha_1}...x_dim^{\alpha_dim}$. The i-th element of the coefficients
+     * vector contains the coefficient for the i-th monomial.
+     */
+    Polynomial (const Table<2,double> &exponents,
+                const Vector<double>   &coefficients,
+                const unsigned int n_components = 1);
```

Comment:
I used n_components because Functions::Monomial is written in the same way, and as they are similar classes I wanted them to be consistent with each other. But It is true that n_components has not meaning at all in the class so I will remove it.